src/http.cc

```diff
@@ -2377,7 +2380,7 @@ HttpStateData::buildRequestPrefix(MemBuf * mb)
     {
         HttpHeader hdr(hoRequest);
         forwardUpgrade(hdr); // before httpBuildRequestHeader() for CONNECTION
-        httpBuildRequestHeader(request.getRaw(), entry, fwd->al, &hdr, flags);
+        httpBuildRequestHeader(request.getRaw(), entry, fwd->al, &hdr, _peer, flags);
```

Comment:
This call does not look safe because httpBuildRequestHeader() expects a valid raw CachePeer pointer (including nil) while this raw `_peer` might become _invalid_ after independent code changes. Let's check pointer validity before passing it to a function that expects valid pointers.

```suggestion
        const auto peer = cbdataValid(_peer) ? _peer : nullptr;
        httpBuildRequestHeader(request.getRaw(), entry, fwd->al, &hdr, peer, flags);
```

Please check other callers as well.

Reply:
Done.